nodeprobe/nodeprobe.go

```diff
@@ -2,164 +2,186 @@ package nodeprobe
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"sync"
-	"sync/atomic"
 	"time"
 
 	"go.uber.org/zap"
 )
 
+type nodeName string
+
 const (
-	probeInterval = 24 * time.Second
+	cl          = "consensus client"
+	el          = "execution client"
+	eventSyncer = "event-syncer"
 )
 
+// Node represents a node being probed.
 type Node interface {
 	Healthy(ctx context.Context) error
 }
 
+const (
+	probeFrequency = 60 * time.Second
+)
+
+// Prober probes(monitors) the nodes it is configured with. It can do it periodically in the background via
+// Start func (which can crash the process via logger.Fatal call if it finds out that one of the nodes being
+// probed is not healthy), or it can probe on demand via Probe func.
+// It is used to make sure the Ethereum nodes (CL, EL) are up and running, and they are healthy enough to for
+// SSV node to be able to perform duties.
 type Prober struct {
-	logger           *zap.Logger
-	interval         time.Duration
-	nodes            map[string]Node
-	nodesMu          sync.Mutex
-	healthy          atomic.Bool
-	cond             *sync.Cond
-	unhealthyHandler func()
+	logger *zap.Logger
+
+	// nodesMu protects access to nodes, needed to handle the case when "event syncer" is added dynamically later on
+	// when the Prober is already running.
+	nodesMu sync.Mutex
+	nodes   map[nodeName]Node
 }
 
-func NewProber(logger *zap.Logger, unhealthyHandler func(), nodes map[string]Node) *Prober {
+func NewProber(logger *zap.Logger, consensusClient, executionClient Node) *Prober {
 	return &Prober{
-		logger:           logger,
-		unhealthyHandler: unhealthyHandler,
-		interval:         probeInterval,
-		nodes:            nodes,
-		cond:             sync.NewCond(&sync.Mutex{}),
+		logger: logger,
+		nodes: map[nodeName]Node{
+			el: executionClient,
+
+			// Underlying options.Beacon's value implements nodeprobe.StatusChecker.
+			// However, as it uses spec's specssv.BeaconNode interface, avoiding type assertion requires modifications in spec.
+			// If options.Beacon doesn't implement nodeprobe.StatusChecker due to a mistake, this would panic early.
+			cl: consensusClient,
+		},
 	}
 }
 
 func (p *Prober) Start(ctx context.Context) {
-	go func() {
-		if err := p.Run(ctx); err != nil {
-			p.logger.Error("finished probing nodes", zap.Error(err))
-		}
-	}()
-}
-
-func (p *Prober) Run(ctx context.Context) error {
-	ticker := time.NewTicker(p.interval)
+	ticker := time.NewTicker(probeFrequency)
 	defer ticker.Stop()
 
 	for {
-		p.probe(ctx)
+		func() {
```

Comment:
Why do we need to wrap it with `func() {} ()`?

Reply:
Anyway, we call `cancel` when it's never used anymore with this wrapping. Indeed, it would be canceled earier, but I doubt it would change the behavior

Reply:
I looked at the other case while writing the above. Actually, that's not true because of the for loop. Alright, I'm resolving this